src/ModAttachr.ts

```diff
@@ -48,12 +54,14 @@ export class ModAttachr implements IModAttachr {
             for (const mod of settings.mods) {
                 this.addMod(mod);
             }
-        }
+        }
+
+        this.eventNames = new EventNames();

```

Comment:
`eventNames` should be allowed as a member of settings, and if not provided default to a new `EventNames`. That way FSP and other consumers can define their own instances of it that override the default.